### ISSUE TITLE
Pokemon Super Mystery Dungeon: Softlock when opening Meowth Theater in multiple cases

### ISSUE THREAD

**Issue by j-selby** [citra-emu/citra#2118]
Commit: cc7f115
Save Game (right at the correct spot): [00174600.zip](https://github.com/citra-emu/citra/files/518130/00174600.zip)

Accessing Meowth's Theater initially writes "Meowth's Theater data" immediately, and dumps you to a menu. Selecting a option from this menu freezes the game with either of the below screenshots depending on what you select. After the first opening of Meowth's Theater you are always thrown to the screenshot as well without seeing the menu. Meowth's Theater data is saved even if you don't choose the main save option from the main menu. During all softlocks, no keys work and graphics is frozen, but audio continues to play. This occurs with both CPU JIT on and off.

Update: Deleting Meowth's Theater data from extdata (00001746) allows us access to the main menu again. It now produces the second screenshot, as attached when attempting to access a dungeon, with another softlock.

_Insert witty Team Rocket joke here._

Log during initial write:

```
[  10.219106] Service.FS <Error> core\hle\service\fs\fs_user.cpp:Service::FS::OpenArchive:435: failed to get a handle for archive archive_id=0x00000006 archive_path=[Binary: 010000004617000000000000]
[  10.220596] Service.FS <Warning> core\hle\service\fs\fs_user.cpp:Service::FS::CreateExtSaveData:628: (STUBBED) savedata_high=00000000 savedata_low=00001746 cmd_buff[3]=00000000 cmd_buff[4]=00030002 cmd_buff[5]=00000064 cmd_buff[6]=00001388 cmd_buff[7]=FFFFFFFF cmd_buff[8]=FFFFFFFF icon_size=000036C0 icon_descriptor=00036C0A icon_buffer=31DF2800
[  10.221065] Common.Filesystem <Warning> common\file_util.cpp:FileUtil::IsDirectory:116: stat failed on C:\Users\James\Downloads\citra-latest-windows-amd64/user/sdmc/Nintendo 3DS/00000000000000000000000000000000/00000000000000000000000000000000/extdata/00000000/00001746/: The system cannot find the file specified.
[  10.221406] Common.Filesystem <Warning> common\file_util.cpp:FileUtil::IsDirectory:116: stat failed on C:\Users\James\Downloads\citra-latest-windows-amd64/user/sdmc/Nintendo 3DS/00000000000000000000000000000000/00000000000000000000000000000000/extdata/00000000/00001746/user/: The system cannot find the file specified.
[  10.222051] Common.Filesystem <Warning> common\file_util.cpp:FileUtil::IsDirectory:116: stat failed on C:\Users\James\Downloads\citra-latest-windows-amd64/user/sdmc/Nintendo 3DS/00000000000000000000000000000000/00000000000000000000000000000000/extdata/00000000/00001746/boss/: The system cannot find the file specified.
```

Log during dungeon access:

```
 [  88.973881] Common.Filesystem <Warning> common\file_util.cpp:FileUtil::IsDirectory:116: stat failed on C:\Users\James\Downloads\citra-latest-windows-amd64/user/sdmc/Nintendo 3DS/00000000000000000000000000000000/00000000000000000000000000000000/extdata/00000000/00001746/user//personal/: The system cannot find the file specified.
[  88.975368] Service.FS <Error> core\hle\service\fs\fs_user.cpp:Service::FS::OpenDirectory:397: failed to get a handle for directory type=4 size=22 data=[Wchar: /personal/]
[  88.977203] Common.Filesystem <Warning> common\file_util.cpp:FileUtil::IsDirectory:116: stat failed on C:\Users\James\Downloads\citra-latest-windows-amd64/user/sdmc/Nintendo 3DS/00000000000000000000000000000000/00000000000000000000000000000000/extdata/00000000/00001746/user//personal/metadata.bin: The system cannot find the path specified.
[  88.977295] Service.FS <Error> core\file_sys\disk_archive.cpp:FileSys::DiskFile::Open:129: Non-existing file C:\Users\James\Downloads\citra-latest-windows-amd64/user/sdmc/Nintendo 3DS/00000000000000000000000000000000/00000000000000000000000000000000/extdata/00000000/00001746/user//personal/metadata.bin can't be open without mode create.
[  88.977389] Service.FS <Error> core\hle\service\fs\fs_user.cpp:Service::FS::OpenFile:76: failed to get a handle for file [Wchar: /personal/metadata.bin]
```

Screenshots:

Accessing after initial creation:
![citra-qt-09-10-2016_17-11-31](https://cloud.githubusercontent.com/assets/1404334/19218365/cfb2d642-8e43-11e6-8124-9e16d543828e.png)

Accessing Meowth dungeon:
![chrome-09-10-2016_17-55-39](https://cloud.githubusercontent.com/assets/1404334/19218607/f007e9cc-8e49-11e6-9832-239ca22e5abc.png)

**Comment by shinyquagsire23** [comment]
Tested on my own save, can confirm same issue. The data in extdata is separate from the main save so that behavior is same as hardware. Deleting extdata manually from the Citra SD card allows re-entry into the menu it seems. 

The log is irrelevant to this issue, though. Those prints are a result of the Treehouse team leaving in debug code to load language files from SD (the game doesn't have SD access though, so it's only usable in Citra or CFW which allows SD access everywhere).

My guess is that this is possibly related to #2101 and/or #2094, perhaps it attempts creation every time and checks for specific error codes?

**Comment by j-selby** [comment]
@shinyquagsire23 Yeah, I figured the log file was irrelevant, but wanted to post it nonetheless. I can confirm that deleting 00001746 from extdata does grant us access to the menu. I'll update the OP in a sec.

**Comment by Subv** [comment]
Does this still happen?

**Comment by j-selby** [comment]
@Subv Still broken, including after removing extdata.

**Comment by adityaruplaha** [comment]
Has this one been fixed?
https://community.citra-emu.org/t/workaround-pokemon-super-mystery-dungeon-crash-locations/1465/12
The above says that these crashes have been fixed.

**Comment by Hexagon12** [comment]
@adityaruplaha The issue in the link is completely different from this issue.

**Comment by adriansg15** [comment]
Hi!! Does anyone know where I can download the game?

**Comment by adityaruplaha** [comment]
Has this been fixed ?

**Comment by B3n30** [comment]
Closed by #3091
